scripts/setup_functions

```diff
@@ -10,16 +10,23 @@ PYTHON_VERSION="${PYTHON_VERSION:-python3.9}"
 BASE_DIR="$(git rev-parse --show-toplevel)"
 
 ensure_virtual_env () {
-  # look for VIRTUAL_ENV variable; create virtualenv in .venv in this directory if not found, or if the path
-  # does not exist (which can happen if a virtualenv folder is deleted while the virtualenv is still activated)
+  # look for VIRTUAL_ENV variable; create virtualenv in .venv in this directory
+  # if not found, or if the path does not exist (which can happen if a
+  # virtualenv folder is deleted while the virtualenv is still activated)
   if [[ -v VIRTUAL_ENV && (! -f $VIRTUAL_ENV) ]] || [[ ! -v VIRTUAL_ENV ]]; then
-    if [[ ! -f "$BASE_DIR/.venv/bin/activate" ]]; then
+    # set activate path here so we can override it with the new path if there's
+    # no existing virtualenv
+    ACTIVATE_PATH="${VIRTUAL_ENV:-""}/bin/activate"
```

Comment:
You don't need the empty string here. The usual way of writing this is just `${VIRTUAL_ENV:-}`.

Reply:
TIL, thank you 😄